packages/core/src/config/promotion/promotion-condition.ts

```diff
@@ -18,6 +19,7 @@ import { Order } from '../../entity/order/order.entity';
  * @docsPage promotion-condition
  */
 export type CheckPromotionConditionFn<T extends ConfigArgs> = (
+    ctx: RequestContext,
```

Comment:
This part with adding the RequestContext to the check function seems to be mixing concerns. Is there some way that this is connected with making Customer channel-aware? If not, I'd suggest you leave them out of this PR, and create a separate issue/PR with some reasoning for it (I think I can guess, but it's good to make such decisions explicit and formal).

Reply:
This is added because the default `CustomerGroupCondition` references `customerService.getCustomerGroups()`. Previously this method did not require a context, but with ChannelAware customers it now does. Therefor a RequestContext needs to be passed to the check method.

Reply:
OK I get it, thanks for clarifying.